RFS/src/main/java/com/rfs/worker/RfsWorker.java

```diff
@@ -0,0 +1,45 @@
+package com.rfs.worker;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+import com.rfs.cms.CmsClient;
+import com.rfs.cms.CmsEntry.Snapshot;
+import com.rfs.cms.CmsEntry.SnapshotStatus;
+import com.rfs.common.SnapshotCreator;
+
+public class RfsWorker {
+    private static final Logger logger = LogManager.getLogger(RfsWorker.class);
+    private final CmsClient cmsClient;
+    private final GlobalData globalState;
+    private final SnapshotCreator snapshotCreator;
+
+    public RfsWorker(GlobalData globalState, CmsClient cmsClient, SnapshotCreator snapshotCreator) {
+        this.globalState = globalState;
+        this.cmsClient = cmsClient;
+        this.snapshotCreator = snapshotCreator;
+    }
+
+    public void run() throws Exception {
+        logger.info("Running RfsWorker");
+
+        while (true) {
```

Comment:
This outer while loop is no longer necessary; a holdover from a previous iteration that I forgot to remove.